brand-ing/study-group-finder
# Frontend setup

Almost ready; needs testing

### TODO
- Remove commented code and/or add edit & delete functions + navbar
- Change extensions added (e.g. currently frontend has gitlens but backend does not)

### Changes (Resolves #2) 
- Changed api endpoint locations
- Added api endpoint for listing all CRUDs
- Configured second devcontainer; devcontainers now use Docker compose
  - **You may need to remove the existing devcontainers using VSCode's Remote Explorer**
- Basic frontend that can view 'CRUDs' from the database
- Changes and fixes to java/gradle configuration
### INSTRUCTIONS (to test integration)

1. Install WSL2 (if using Windows) 
2. Install Docker Desktop 
3. Ensure you have a serviceAccountKey.json file in \backend\secret\ (from firebase console)
Note: Might need to re-start the app and Docker Engine whenever you restart your computer
4. Open root project workspace folder or `study-group-finder.code-workspace`
5. Press F1 -> "Devcontainers: Reopen Folder in Container..." -> Navigate to `frontend` or `backend` folder, and then open
Note: This may start both containers at once. If you don't see two VSCode windows, open a new Window and repeat steps 4-5 to open the other devcontainer. 
6. In the VSCode window for the backend container, navigate to the Springboot Dashboard and run the backend app
Note: If you don't see any apps, you may need to wait for the project to finish configuration.
7. In the window for the frontend container, navigate to /frontendapp 
8. If this is the first time, run `npm install` (node_modules should be cached)
9. Run `npm start` and wait for the frontend to start up. VSCode should automatically open a browser tab for http://localhost:3000/. Click the 'cruds' button.


### Expected Result
![image](https://github.com/user-attachments/assets/bc91373c-fb5d-4f2f-9366-e2d62d9e0aa8)

### Known issues or complications
1. Devcontainers use a lot of memory
This may not necessarily be an issue as some memory should be released back to the Host OS if necessary. You can also reduce active memory usage by closing one of the windows and stopping one of the containers.
2. Folders deleted inside of Devcontainers are **gone forever**
Pay attention to VSCode warnings and use commits to save work (you can squash them later) or maybe something like `git stash push --keep-index`

### Based on
The CRUD functionality is based on https://www.baeldung.com/spring-boot-react-crud and  https://www.youtube.com/watch?v=clH7SxG-Vdc&list=PLS1QulWo1RIYMQcf1y2bqQZbpXLzpKDUL with several modifications for compatibility.

If you have any questions, please ask.
